test/integration/component/test_volumes.py

```diff
@@ -603,7 +603,134 @@ def test_01_volume_attach_detach(self):
                 "Check the state of VM"
             )
         except Exception as e:
-            self.fail("Exception occuered: %s" % e)
+            self.fail("Exception occurred: %s" % e)
+        return
+    
+    @attr(tags=["advanced", "advancedns"])
+    def test_02_root_volume_attach_detach(self):
+        """Test Root Volume attach/detach to VM
+        """
+
+        # Validate the following
+        # 1. Deploy a VM
+        # 2. Check for root volume
+        # 3. Stop VM
+        # 4. Detach root volume
+        # 5. Verify root volume detached
+        # 6. Attach root volume
+        # 7. Start VM
+        
+        try:
+            # Check for root volume
```

Comment:
@shwetaag Thanks for the feedback.  You are absolutely right, I should test to ensure the hypervisor supports Root Volume detach.  Root Volume detach is currently supported by a number of hypervisors, so I will put in an if block that will make sure that it only tests the supported hypervisors.